### DISCUSSION TITLE
API prefix

### DISCUSSION THREAD

**Discussion by Magnuti** [pocketbase/pocketbase#2453] [Q&A] [answered]
Is it correct that the endpoints defined within `app.OnBeforeServe` become defined on root and not under the `/app` prefix?

Running the example code in the README https://github.com/pocketbase/pocketbase/blob/c6d599244239ed17b2f2f7ce892b1279ddabf5ac/README.md?plain=1#LL69C13-L69C13 that specifies `Path:   "/hello",` causes the endpoint `/api/hello` to return 404, but the endpoint `/hello` to return OK.

```
$ curl http://0.0.0.0:8090/api/health
{"code":200,"message":"API is healthy."}
```

```
$ curl http://0.0.0.0:8090/api/hello 
{"code":404,"message":"Not Found.","data":{}}
```

```
$ curl http://0.0.0.0:8090/hello    
Hello world!
```
The [router docs](https://pocketbase.io/docs/router) mentions that *To avoid collisions with future internal routes it is recommended to use a unique prefix when adding routes to the /api base endpoint, eg. /api/yourapp/... path.*.

Should "api/yourapp/... path" be changed to "yourapp/... path" ?

**Comment by ganigeorgiev** [comment, marked as answer]
> Is it correct that the endpoints defined within app.OnBeforeServe become defined on root and not under the /app prefix?

Yes. It is up to the developers to choose a route prefix (if at all). An example situation where you may want a root based route is serving/rendering custom static files/pages (eg. `pb_public`).

> Should "api/yourapp/... path" be changed to "yourapp/... path" ?

The note in the docs is only when you want to add routes under the `/api` base endpoint. I'll consider updating the note to avoid the confusion with the planned docs refactoring in the near future.
